### PR TITLE
docs(text-field): Add HelperText Validity Usage Docs

### LINKED ISSUE #177: Add HelperText Usage Docs for use with validation prop
Hello,

I'm trying to get HelperText component to display as an error message like in material.io specs but I really don't know what props to set and when.

Also, if I set both `isValid` and `validation` to true, the helper text doesn't appear.
It's opacity is set to 0 (expected) but the transition isn't being run when I press the text field.

### Step to reproduce
Do the usual `create-react-app` setup and `npm install @material/helper-text`.
Import the sass file:
``` css
@import '@material/react-text-field/index.scss';
```
And use this App.js
``` js
import React, { Component } from 'react';
import TextField, {Input, HelperText} from '@material/react-text-field';
import './App.css';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {value:"hello"};
  }

  render() {
    return (
      <div className="App">
        <TextField
          box
          label='Dog'
          helperText={<HelperText isValid validation>Help Me!</HelperText>}
        >
          <Input
            value={this.state.value}
            onChange={(e) => this.setState({value: e.target.value})}/>
        </TextField>
      </div>
    );
  }
}

export default App;
```

### Expected behaviour
The helper text to appear, ideally in a different color.

### What actually happens
Nothing, the text helper doesn't appear.

### How I'll handle it for now

I'll simply use it without setting any props and do the color logic on my side but it'd be helpful if the component did this by itself (trembling animations and all), since it has the proper props.

But maybe I'm I doing something wrong ?
Maybe I forgot to include some scss files or I'm missing a character somewhere ?

### PR DESCRIPTION
**Pull request by Ranguna**
Fixes #177

### CODE REVIEW

File: packages/text-field/helper-text/README.md [hunk 1]
```diff
@@ -30,6 +30,39 @@ role | String | Sets role on HTML element.
 showToScreenReader | Boolean | toggles the aria-hidden property to allow element to be visible to the screen reader.
 validation | Boolean | If true, alters the helper text to an error message.
 
+## Input Validation
+
+HelperText provides validity styling by setting the `validation` prop in HelperText. Validation can be checked through the appropriate `Input` validation properties (pattern, min, max, require, step, minlength, maxlength).
```
Comment (moog16) [inline packages/text-field/helper-text/README.md]: require --> required

Comment (moog16) [reply packages/text-field/helper-text/README.md]: capitalize minlength/maxlength --> minLength/maxLength

File: packages/text-field/helper-text/README.md [hunk 2]
```diff
@@ -30,6 +30,39 @@ role | String | Sets role on HTML element.
...
+
+The following snipped is an example of how to use regex and check for minimum length with HelperText:
```
Comment (moog16) [inline packages/text-field/helper-text/README.md]: - snipped --> snippet
- regex --> pattern regex

File: packages/text-field/helper-text/README.md (hunk 1 above)
Comment (Ranguna) [reply packages/text-field/helper-text/README.md]: Should have reviewed those props, will change.

File: packages/text-field/helper-text/README.md (hunk 2 above)
Comment (Ranguna) [reply packages/text-field/helper-text/README.md]: How did the `d` get in there, it's so far apart from the `t`.
I'll change that too.

**Comment by Ranguna** [comment]
I signed it!

**Comment by moog16** [comment]
Just a docs change, so ignoring tests

**Comment by Ranguna** [comment]
No problem, glad I could help :+1:

**Pull request merged**